README.md

```diff
@@ -52,34 +52,19 @@ interface Promise
     /**
      * Registers a callback to be invoked when the promise is resolved.
      *
-     * The callback receives `null` as first parameter and `$value` as second parameter on success. It receives the
-     * failure reason as first parameter and `null` as second parameter on failure.
-     *
      * If the promise is already resolved, the callback MUST be executed immediately.
      *
-     * Warning: If you use type declarations for `$value`, be sure to make them accept `null` in case of failures.
-     *
-     * @param callable(\Throwable|\Exception|null $exception, mixed $value) $onResolved Callback to be executed.
+     * @param callable(mixed $reason, mixed $value) <`$reason` shall be `null` on success, `$value` shall be `null` on failure>
```

Comment:
@krakjoe It's the route we took everywhere. It's the way it's recommended by phpdoc.org. It's the usual way you'll find it in existing projects.

If you want to argue about something here, then argue about typed callables, please.